File: back/models/groups/threadMessageModel.js
const mongoose = require('mongoose');

const threadMessageSchema = new mongoose.Schema({
  threadId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Thread',
    required: true
  },
  senderId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  senderName: {
    type: String,
    required: true
  },
  senderImageUrl: {
    type: String,
    default: null
  },
  message: {
    type: String,
    required: true
  },
  isDeleted: {
    type: Boolean,
    default: false
  },
  readBy: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  }]
}, { timestamps: true });

module.exports = mongoose.model('ThreadMessage', threadMessageSchema);

File: front/lib/model/groups/thread_message_model.dart
class ThreadMessage {
  final String id;
  final String threadId;
  final String senderId;
  final String senderName;
  final String? senderImageUrl;
  final String message;
  final bool isDeleted;
  final List<String> readBy;
  final DateTime createdAt;
  final DateTime updatedAt;

  ThreadMessage({
    required this.id,
    required this.threadId,
    required this.senderId,
    required this.senderName,
    this.senderImageUrl,
    required this.message,
    required this.isDeleted,
    required this.readBy,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ThreadMessage.fromJson(Map<String, dynamic> json) {
    return ThreadMessage(
      id: json['_id'] ?? '',
      threadId: json['threadId'] ?? '',
      senderId: json['senderId']?['_id'] ?? json['senderId'] ?? '',
      senderName: json['senderName'] ?? 'Unknown',
      senderImageUrl: json['senderImageUrl'],
      message: json['message'] ?? '',
      isDeleted: json['isDeleted'] ?? false,
      readBy: List<String>.from(json['readBy'] ?? []),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}
